lib/rules/forbid-foreign-prop-types.js

```diff
@@ -34,15 +48,56 @@ module.exports = {
       return node.parent.type === 'AssignmentExpression' && node.parent.left === node;
     }
 
+    function findParentAssignmentExpression(node) {
+      let parent = node.parent;
+
+      while (parent && parent.type !== 'Program') {
+        if (parent.type === 'AssignmentExpression') {
+          return parent;
+        } else if (parent.parent) {
+          parent = parent.parent;
+        } else {
+          return false;
```

Comment:
i think we can collapse the else if and the else together, and just do `parent = parent.parent`, and let the while condition catch it? (also, maybe `null` would make more sense than `false`)